datasets/nlu_evaluation_data/nlu_evaluation_data.py

```diff
@@ -0,0 +1,234 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""NLU Evaluation Data."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import re
+
+import datasets
+
+
+logger = datasets.logging.get_logger(__name__)
+
+
+_CITATION = """\
+@InProceedings{XLiu.etal:IWSDS2019,
+  author    = {Xingkun Liu, Arash Eshghi, Pawel Swietojanski and Verena Rieser},
+  title     = {Benchmarking Natural Language Understanding Services for building Conversational Agents},
+  booktitle = {Proceedings of the Tenth International Workshop on Spoken Dialogue Systems Technology (IWSDS)},
+  month     = {April},
+  year      = {2019},
+  address   = {Ortigia, Siracusa (SR), Italy},
+  publisher = {Springer},
+  pages     = {xxx--xxx},
+  url       = {http://www.xx.xx/xx/}
+}
+"""
+
+# You can copy an official description
+_DESCRIPTION = """\
+Raw part of NLU Evaluation Data. It contains 25 715 non-empty examples from 68 unique intents belonging to 18 scenarios.
+"""
+
+_HOMEPAGE = "https://github.com/xliuhw/NLU-Evaluation-Data"
+
+_LICENSE = "Creative Commons Attribution 4.0 International License (CC BY 4.0)"
+
+_URL = "https://raw.githubusercontent.com/xliuhw/NLU-Evaluation-Data/master/AnnotatedData/NLU-Data-Home-Domain-Annotated-All.csv"
+
+ANNOTATION_PATTERN = re.compile(r"\[(.+?)\s+\:+\s(.+?)\]")
+
+
+def remove_annotations(text):
+    """
+    Remove named entity annotations from text example.
+
+    Examples are defined based on `answer_annotation` column since it has the least number
+    of Nans. However, this column contains patterns of annotation of the form:
+
+        [named_entity : part_of_text]
+
+        e.g. [time : five am], [date : this week]
+
+    We identity them with regex rule and replace all occurrences with just part_of_text.
+    """
+    return ANNOTATION_PATTERN.sub(r"\2", text)
+
+
+def define_intent_name(scenario, intent):
+    """
+    Intent name is defined as concatenation of `scenario` and `intent` values.
+
+    See Also:
+        https://github.com/xliuhw/NLU-Evaluation-Data/issues/5
+    """
+    return f"{scenario}_{intent}"
+
+
+class NLUEvaluationDataset(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class NLUEvaluationData(datasets.GeneratorBasedBuilder):
```
This must be the camel case version of the dataset script name `nlu_evaluation_data`